psdk_wrapper/src/psdk_wrapper.cpp

```diff
@@ -232,40 +236,54 @@ PSDKWrapper::set_environment()
+  return_code = DjiUserConfigManager_LoadConfiguration(
+      params_.link_config_file_path.c_str());
   if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
   {
     RCLCPP_ERROR(get_logger(),
-                 "Register HAL USB BULK handler error. Error code is: %ld",
+                 "Configuration file could not be loaded. Error code is: %ld",
                  return_code);
     return false;
   }
-#elif (HARDWARE_CONNECTION == DJI_USE_UART_AND_NETWORK_DEVICE)
-  RCLCPP_INFO(get_logger(), "Using DJI_USE_UART_AND_NETWORK_DEVICE");
-  T_DjiHalNetworkHandler network_handler;
-  network_handler.NetworkInit = HalNetWork_Init;
-  network_handler.NetworkDeInit = HalNetWork_DeInit;
-  network_handler.NetworkGetDeviceInfo = HalNetWork_GetDeviceInfo;
-  return_code = DjiPlatform_RegHalNetworkHandler(&network_handler);
-  if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+  RCLCPP_INFO(get_logger(), "Loaded configuration file");
+  DjiUserConfigManager_GetLinkConfig(&linkConfig);
+  if (linkConfig.type == DJI_USER_LINK_CONFIG_USE_UART_AND_USB_BULK_DEVICE)
+  {
+    RCLCPP_INFO(get_logger(), "Using DJI_USE_UART_USB_BULK_DEVICE");
+    T_DjiHalUsbBulkHandler usb_bulk_handler;
+    usb_bulk_handler.UsbBulkInit = HalUsbBulk_Init;
+    usb_bulk_handler.UsbBulkDeInit = HalUsbBulk_DeInit;
+    usb_bulk_handler.UsbBulkWriteData = HalUsbBulk_WriteData;
+    usb_bulk_handler.UsbBulkReadData = HalUsbBulk_ReadData;
+    usb_bulk_handler.UsbBulkGetDeviceInfo = HalUsbBulk_GetDeviceInfo;
+    return_code = DjiPlatform_RegHalUsbBulkHandler(&usb_bulk_handler);
+    if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+    {
+      RCLCPP_ERROR(get_logger(),
+                   "Register HAL USB BULK handler error. Error code is: %ld",
+                   return_code);
+      return false;
+    }
+  }
+  else if (linkConfig.type == DJI_USER_LINK_CONFIG_USE_UART_AND_NETWORK_DEVICE)
   {
-    RCLCPP_ERROR(get_logger(),
-                 "Register HAL Network handler error. Error code is: %ld",
-                 return_code);
+    RCLCPP_INFO(get_logger(), "Using DJI_USE_UART_AND_NETWORK_DEVICE");
+    T_DjiHalNetworkHandler network_handler;
+    network_handler.NetworkInit = HalNetWork_Init;
+    network_handler.NetworkDeInit = HalNetWork_DeInit;
+    network_handler.NetworkGetDeviceInfo = HalNetWork_GetDeviceInfo;
+    return_code = DjiPlatform_RegHalNetworkHandler(&network_handler);
+    if (return_code != DJI_ERROR_SYSTEM_MODULE_CODE_SUCCESS)
+    {
+      RCLCPP_ERROR(get_logger(),
+                   "Register HAL Network handler error. Error code is: %ld",
+                   return_code);
+    }
+  }
+  else
+  {
+    RCLCPP_INFO(get_logger(), "Using DJI_USE_ONLY_UART");
   }
-#elif (HARDWARE_CONNECTION == DJI_USE_ONLY_UART)
-  RCLCPP_INFO(get_logger(), "Using DJI_USE_ONLY_UART");
-  /*!< Attention: Only use uart hardware connection.
-   */
-#endif
 
   // Attention: if you want to use camera stream view function, please uncomment
   // it.
```

Comment:
this comment seems old, is it referring to something meaningful?

Reply:
removed